subsys/storage/flash_map/flash_map_default.c

```diff
@@ -19,26 +17,48 @@
 	 .fa_dev = DEVICE_DT_GET(DT_MTD_FROM_FIXED_PARTITION(part)),	        \
 	 .fa_size = DT_REG_SIZE(part),						\
 	 .fa_label = DT_PROP_OR(part, label, NULL),	},
+#define FLASH_AREA_SUB_FOO(part)						\
```

Comment:
These (now 4) macros are all doing the same thing with minor variations, can't they be compressed into a single version?
```
#define FLASH_AREA_FOO(part, mtd_from_partition)							\
	{.fa_id = DT_FIXED_PARTITION_ID(part),					\
	 .fa_off = DT_REG_ADDR(part),						\
	 .fa_dev = DEVICE_DT_GET(mtd_from_partition(part)),	        \
	 .fa_size = DT_REG_SIZE(part),						\
	 COND_CODE_1(IS_ENABLED(CONFIG_FLASH_MAP_LABELS), (.fa_label = DT_PROP_OR(part, label, NULL),), ()) },
...
FLASH_AREA_FOO(part, DT_MTD_FROM_FIXED_PARTITION)
FLASH_AREA_FOO(part, DT_MTD_FROM_FIXED_SUBPARTITION)
```